tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -0,0 +1,301 @@
+        width: 100%;
+      }
+      .legend-row {
+        margin: 5px 0 5px 0;
+        width: 100%;
+      }
+      .run-label {
+        width: 30%;
+        display: inline-block;
+        word-wrap: break-word;
+        box-sizing: border-box;
+        padding: 0 0 0 8px;
+      }
+      .color-box {
+        display: inline-block;
+        border-radius: 1px;
+        width: 10px;
+        height: 10px;
+      }
+      .step-label-text {
+        font-weight: bold;
+      }
+      .wall-time-label-text {
+        color: #888;
+        font-size: 0.8em;
+      }
+    </style>
+  </template>
+  <script>
+    "use strict";
+    import {Canceller} from "../tf-backend/canceller.js";
+    import {getRouter} from "../tf-backend/router.js";
+    import {runsColorScale} from "../tf-color-scale/colorScale.js";
+    import * as ChartHelpers from '../vz-line-chart/vz-chart-helpers.js';
+
+    Polymer({
+      is: "tf-pr-curve-card",
+      properties: {
+        runs: Array,
+        tag: String,
+        // For each run, the card will display the PR curve at this step or the one
+        // closest to it, but less than it.
+        runToStepCap: Object,
+        _runToPrCurveEntry: {
+          type: Object,
+          computed: "_computeActualPrCurveEntryPerRun(_runToData, runToStepCap)",
+        },
+        // A list of runs with an available step. Used to populate the table of
+        // steps per run.
+        _runsWithStepAvailable: {
+          type: Array,
+          computed: "_computeRunsWithStepAvailable(runs, _runToPrCurveEntry)",
+        },
+        _runToData: Object,
+        _colorScaleFunction: {
+          type: Object,  // function: string => string
+          value: () => ({scale: runsColorScale}),
+        },
+        requestManager: Object,
+        _canceller: {
+          type: Object,
+          value: () => new Canceller(),
+        },
+        _attached: Boolean,
+        // The value field is a function that returns a function because Polymer
+        // will actually call the value field if the field is a function. However,
+        // we actually want the value itself to be a function.
+        _xComponentsCreationMethod: {
+          type: Object,
+          readOnly: true,
+          value: () => (() => {
+            const scale = new Plottable.Scales.Linear();
+            return {
+              scale: scale,
+              axis: new Plottable.Axes.Numeric(scale, 'bottom'),
+              accessor: d => d.recall,
+            };
+          }),
+        },
+        _yValueAccessor: {
+          type: Object,
+          readOnly: true,
+          // This function returns a function because polymer calls the outer
+          // function to compute the value. We actually want the value of this
+          // property to be the inner function.
+          value: () => (d => d.precision),
+        },
+        _tooltipColumns: {
+          type: Array,
+          readOnly: true,
+          value: [
```

Comment:
If you use the raw array, then the array will be shared across all instances of the chart. Any accidental mutations will be propagated, so you can have state changes coming when you don't expect them. If your code isn't buggy, there's no problem, but if it is buggy then it can be much harder to debug.

With the function initializer, a new array is created for each instance.